Objects/obmalloc.c

```diff
@@ -3169,6 +3169,9 @@ free_obmalloc_arenas(PyInterpreterState *interp)
     for (uint i = 0; i < maxarenas; ++i) {
         // free each obmalloc memory arena
         struct arena_object *ao = &allarenas[i];
+        /* Skip arenas which are not allocated. */
+        if (ao->address == (uintptr_t)NULL)
+            continue;
```

Comment:
It may make more sense to do this in `_PyMem_ArenaFree()`.

Reply:
Also,

```suggestion
        /* Skip arenas which are not allocated. */
        if (ao->address == (uintptr_t)NULL) {
            continue;
        }
```

Reply:
OK, happy to move it there.